src/blocks/custom.rs

```diff
@@ -0,0 +1,117 @@
+use std::time::{Duration, Instant};
+use std::process::Command;
+use std::error::Error;
+use std::iter::{Cycle, Peekable};
+use std::vec;
+use std::sync::mpsc::Sender;
+
+use block::Block;
+use widgets::button::ButtonWidget;
+use widget::I3BarWidget;
+use input::I3barEvent;
+use scheduler::Task;
+
+use serde_json::Value;
+use uuid::Uuid;
+
+pub struct Custom {
+    id: String,
+    update_interval: Duration,
+    output: ButtonWidget,
+    command: Option<String>,
+    on_click: Option<String>,
+    cycle: Option<Peekable<Cycle<vec::IntoIter<String>>>>,
+    tx_update_request: Sender<Task>,
+}
+
+impl Custom {
+    pub fn new(config: Value, tx: Sender<Task>, theme: Value) -> Custom {
+        let mut custom = Custom {
```

Comment:
I think using  [Field Init Shorthand](https://doc.rust-lang.org/nightly/book/second-edition/appendix-07-newest-features.html) would be better here (stable since 1.17). No need to make custom mutable for initialization.